default/python/AI/pathfinding.py

```diff
@@ -0,0 +1,134 @@
+from heapq import heappush, heappop
+from collections import namedtuple
+
+import freeOrionAIInterface as fo
+from turn_state import state
+
+from common.configure_logging import convenience_function_references_for_logger
+(debug, info, warn, error, fatal) = convenience_function_references_for_logger(__name__)
+
+
+# TODO Allow short idling in system to refuel
```

Comment:
I was talking about the refueling mechanics if the fleet is stationary in an unsupplied system. I clarified that comment.

If our path reaches the target system but is e.g. 0.1 fuel off, then it could be prefered to let it wait there for one turn rather than having to take a detour. The logic to find suitable circumstances isn't trivial and needs some thought but it would be useful in (various) edge cases especially for those living organic ships.